code/lighting/lighting_profiles.cpp

```diff
@@ -0,0 +1,166 @@
+{
+	name = "";
+
+    tonemapper = tnm_Uncharted;
+
+	ppc_values.toe_strength = 0.5f;
+    ppc_values.toe_length = 0.5f;
+    ppc_values.shoulder_strength = 0.0f;
+    ppc_values.shoulder_length = 0.5f;
+    ppc_values.shoulder_angle = 0.1f;
+
+	exposure = 4.0f;
+}
+
+TonemapperAlgorithm lighting_profile::name_to_tonemapper(SCP_string &name)
+{
+	SCP_tolower(name);
+	TonemapperAlgorithm r = tnm_Invalid;
+	if(name == "uncharted" || name == "uncharted 2" ){
+		r = tnm_Uncharted;
+	}
+	if(name == "linear"){
+		r = tnm_Invalid;
+	}
+	if(name == "aces"){
+		r = tnm_Invalid;
+	}
+	if(name == "aces approximate"){
+		r = tnm_Invalid;
+	}
+	if(name == "cineon"){
+		r = tnm_Invalid;
+	}
+	if(name == "cineon"){
+		r = tnm_Invalid;
+	}
+	if(name == "reinhard jodie"){
+		r = tnm_Invalid;
+	}
+	if(name == "reinhard extended"){
+		r = tnm_Invalid;
+	}
+	if(name == "PPC"){
+		r = tnm_Invalid;
+	}
+	if(name == "PPC RGB"){
+		r = tnm_Invalid;
+	}
+	return r;
+}
+
+lighting_profile lighting_profile::default_profile;
+
+void lighting_profile::load_profiles()
+{
+	parse_all();
+}
+
+//The logic for grabbing all the parseable files
+
+void lighting_profile::parse_all()
+{
+	default_profile.reset();
+	if (cf_exists_full("lighting_profiles.tbl", CF_TYPE_TABLES)){
+		mprintf(("TABLES:Starting parse of lighting profiles.tbl"));
+		lighting_profile::parse_file("lighting_profiles.tbl");
+	}
+
+	mprintf(("TBM  =>  Starting parse of lighting profiles ...\n"));
+	parse_modular_table("*-ltp.tbm", lighting_profile::parse_file);
+}
+
+void lighting_profile::parse_default_section()
+{
+	bool keep_going = true;
+	SCP_string buffer;
+	TonemapperAlgorithm tn;
+	while(!optional_string("#END DEFAULT PROFILE") && keep_going){
+		keep_going = false;
+		if(optional_string("$Tonemapper:")){
+			stuff_string(buffer,F_NAME);
+			tn = lighting_profile::name_to_tonemapper(buffer);
+			default_profile.tonemapper = tn;
+			keep_going = true;
+		}
+		keep_going |= optional_parse_into_float("$PPC Toe Strength:",&default_profile.ppc_values.toe_strength);
+		keep_going |= optional_parse_into_float("$PPC Toe Length:",&default_profile.ppc_values.toe_length);
+		keep_going |= optional_parse_into_float("$PPC Shoulder Length:",&default_profile.ppc_values.shoulder_length);
+		keep_going |= optional_parse_into_float("$PPC Shoulder Strength:",&default_profile.ppc_values.shoulder_strength);
+		keep_going |= optional_parse_into_float("$PPC Toe Strength:",&default_profile.ppc_values.shoulder_angle);
+		keep_going |= optional_parse_into_float("$Exposure:",&default_profile.exposure);
+		//TODO: Handle case when there's no line matched but we haven't hit an #end
+		Assert(keep_going);
+	}
+}
+
+//Handle an individual file.
+void lighting_profile::parse_file(const char *filename)
+{
+	try
+	{
+		if (filename == nullptr){
+			//Normal parsing uses this:
+			//read_file_text_from_default(defaults_get_file("lighting_profiles.tbl"));
+			//To set the defaults, I intend to set them as virtual table lines before here
```

Comment:
Is this still correct?
It seems to me like the defaults are applied by just using the reset.

Reply:
That's largely leftover from a fancier branch, will make sure to revise.